### ISSUE TITLE
Bedding point automation?

### ISSUE THREAD

**Issue by RichardScottOZ** [Loop3D/map2loop-2#79]
Anyone tinkering here?

One of the geos expressed interest and I said can automatically make project models.... but only if we have this part.

**Comment by lachlangrose** [comment]
What do you mean by bedding point automation?

**Comment by RichardScottOZ** [comment]
Approximate from other data layers...as generally nowhere will have that field data recorded.

**Comment by lachlangrose** [comment]
Oh I understand. In my opinion, it would make more sense to integrate other datasets into implicit modelling and then get the bedding/orientation data from that. 

What type of datasets are you looking to use?

**Comment by RichardScottOZ** [comment]
Any we can access that will routinely cover whole states (or can be made to).

**Comment by lachlangrose** [comment]
We'd have to see how the particular dataset would be used by a geologist to estimate the bedding orientation. It would be different for different datasets. 

We already use the DEM and the interpreted geology which are usually available everywhere. This is enough information to make a starting model (provided there is enough resolution in the map to show different stratigraphic units to model).

**Comment by RichardScottOZ** [comment]
Ok - I haven't looked at this for a while - does this mean it has been changed to not fail - and produce a basic model if no bedding data exists at all in the particular area selected?

**Comment by lachlangrose** [comment]
I don't know if map2loop will fail if there is no bedding data, but I think it might @markjessell probably be able to help with that. 

I am just thinking about how to do it conceptually. I'm not sure how many use cases there are for it - I think the results would be very uncertain but it depends on what you are going to do with the point estimates.

**Comment by RichardScottOZ** [comment]
Yes, would be but I think the writeups on these things pretty much say that already.

Basically, pick a point, build a basic 3D model to throw in a project.  Exploration projects are also highly uncertain, so that doesn't matter either, in that sense.

e.g. a geologist asks me if they can use map2loop - I say 'no, not without fieldwork giving you this info'.  Which will never be done, generally speaking, so will never be used, apart from if it happens to intersect on an extreme longshot one of the tiny areas it exists already.

**Comment by RichardScottOZ** [comment]
Conceptually, yes - that is the trick - I think there might be some planetary science tricks perhaps that might be useful.

**Comment by markjessell** [comment]
At the moment map2loop is hardwired to require at least 2 or 3 bedding measurements, but if you were willing to accept a model where, for example, all contacts were assumed to be dipping at 45 degrees, I could bypass this. I’ll have a go when Xmas meeting season dies down and see what that gives.



Mark Jessell
Professor/ Western Australian Fellow
Centre for Exploration Targeting/ SES • M006, Perth WA 6009 Australia
T +61 8 6488 5803 • E ***@***.***

________________________________

**Comment by RichardScottOZ** [comment]
Mark,

Thanks - yes, maybe an option for this to toggle?  Whatever you think would suit here in your professional opinion as a generic choice.  That would be great.

Also be happy to try and work something out to run many of them on AWS or something if you wanted to try many different numbers.

I need to talk to some people about what they think re: calculations of such from other things.

**Comment by markjessell** [comment]
Ok. Shouldn’t be too hard to modify the code, just need some quiet time. Would also be curious to get your take on the tomofastx code.

Mark Jessell
Professor/ Western Australian Fellow
Centre for Exploration Targeting/ SES • M006, Perth WA 6009 Australia
T +61 8 6488 5803 • E ***@***.***

________________________________

**Comment by RichardScottOZ** [comment]
This? https://github.com/TOMOFAST/Tomofast-x

Had not seen that one, so thanks.  Was talking to someone Friday re: doing that sort of things with point inversions.

I'd say the 'run on laptops' would have some caveats re: Windows given the gcc and mpi declarations.  I can give it a shot though.

**Comment by markjessell** [comment]
yes that is the one, I would start with WSL as that is less painful!

Mark Jessell
Professor/ Western Australian Fellow
Centre for Exploration Targeting/ SES • M006, Perth WA 6009 Australia
T +61 8 6488 5803 • E ***@***.***

________________________________

**Comment by RichardScottOZ** [comment]
Yeah - would probably try on an EC2 machine first, see how that goes.

**Comment by markjessell** [comment]
Yeah that makes sense

Mark Jessell
Professor/ Western Australian Fellow
Centre for Exploration Targeting/ SES • M006, Perth WA 6009 Australia
T +61 8 6488 5803 • E ***@***.***

________________________________

**Comment by RichardScottOZ** [comment]
The example ran on a compute heavy machine (non parallel) in seconds, anyway.

This one: Parfile_mansf_slice_wavelet.txt

**Comment by RichardScottOZ** [comment]
Did you manage to get to automagic bedding @markjessell  ?

**Comment by markjessell** [comment]
Hi Richard

No, but I haven't tried either! It will need a student to have the time I suspect.

Cheers

Mark

**Comment by RichardScottOZ** [comment]
Speaking of various students, anyone have anything clever here?
